### ISSUE TITLE
find: Could not find the element you were looking for

### ISSUE THREAD

**Issue by impankratov** [getsaf/shallow-render#18]
Hi, thanks for a great lib, I really considering to use it for my projects. Now to the actual question:

when I try to query some element in my test:
```
expect(find('p')).toBeFalsy();
```
test throws an error:
```
context.js:243 Unhandled Promise rejection: Could not find the element you were looking for. Your test tried to access the 'prototype' property on a QueryResult but your query had no results. ; Zone: ProxyZone ; Task: Promise.then ; Value: Error: Could not find the element you were looking for. Your test tried to access the 'prototype' property on a QueryResult but your query had no results.
    at new NoMatchesError (VM40572 query-match.js:16)
    at throwErrorIfNotOneMatch (VM40572 query-match.js:31)
    at Object.getPrototypeOf (VM40572 query-match.js:71)
    at Function.[Symbol.hasInstance] (<anonymous>)
    at Proxy.Object.toString (VM267 zone.js:1466)
    at exports.instanceOf (VM23 context.js:90)
    at ContextKarma.stringify (VM23 context.js:39)
    at ContextKarma.log (VM23 context.js:138)
```

I'm kind of used to do this technique with `fixture.debugElement.query` to make sure that some elements are **not** rendered in some cases. Don't you think that throwing an error in `#find` is a little bit too much? Or I just should stick to using `fixture.debugElement.query` in this case?

**Comment by getsaf** [comment]
The short answer here is to use something like:
```typescript
expect(find('p').length).toBe(0);
// or
expect(find('p')[0]).toBeFalsy();
```

A buddy of mine suggested that I include some custom Jasmine matchers to improve readability. I'd love to do that (or accept a PR if someone else helped). Would be nice to write specs like:

```typescript
expect(find('p')).not.toHaveBeenFound(); // Must be 0 results
expect(find('p')).toHaveFound(1); // Must be exactly 1 result
expect(find('p')).toHaveBeenFound(); // Same as to toHaveFound(1)
```
If you're *really* interested, here's the history behind that error message.

The reason behind the error is that the query results are designed to be used as either a group of results or a single result based upon how it's used in the spec.

For example:
```typescript
it('renders foo in a label', async () => {
  const {find} = await shallow.render();
  const found = find('label');

  expect(found.nativeElement.innerText.toBe('Foo');
   // Throws the NoMatchesError if nothing was found. If we didn't throw the error, the resulting error 
  // would be pretty confusing "Cannot read property 'innerText' of undefined". I figured throwing a 
  // specific NoMatchesError was a better option because it is more informative.
});
```
I wrote a quick spec to see what happens when I try to support `toBeFalsy` and things got a little weird. Jasmine is checking all kinds of properties that don't exist. You're welcome to give it a go to specifically support `toBeFalsy` if you have some ideas, here's the branch with a failing spec:

https://github.com/getsaf/shallow-render/tree/support-to-be-falsy-check

Sorry for the ranting, thanks for your support! I'll leave this open until I get around to supplying some jasmine matchers or we come up with something better.

**Comment by getsaf** [comment]
@impankratov Coming in the next release:
https://github.com/getsaf/shallow-render/pull/20

**Comment by getsaf** [comment]
Closed by #20
